lib/salus/scanners/osv/base.rb

```diff
@@ -0,0 +1,137 @@
+require 'salus/scanners/base'
+require 'json'
+require 'zip'
+require 'uri'
+require 'net/http'
+
+module Salus::Scanners::OSV
+  class Base < Salus::Scanners::Base
+    DATABASE_STRING_MAPPING = {
+      "GHSA" => "Github Advisory Database",
+      "PYSEC" => "Python Packaging Advisory Database",
+      "GO" => "Go Vulnerability Database",
+      "RUSTSEC" => "RustSec Advisory Database",
+      "default" => "Open Source Vulnerabilitiy"
+    }.freeze
+
+    def run
+      raise NoMethodError, 'implement in subclass'
+    end
+
+    def should_run?
+      raise NoMethodError, 'implement in subclass'
+    end
+
+    def name
+      self.class.name.sub('Salus::Scanners::OSV::', '')
+    end
+
+    private
+
+    def osv_vulnerabilities
+      @osv_vulnerabilities ||= fetch_vulnerabilities
+    end
+
+    def osv_urls
+      urls = []
+      urls.append(osv_url_for("Go")) if @repository.go_sum_present? || @repository.go_mod_present?
```

Comment:
if @repository.go_sum_present? || @repository.go_mod_present? differers from the Go osv scanner should_run? logic.

As setup this will download all of the OSV datasets that the repo has lock files for.  The repo config, however may not have a given scanner enabled (pypi osv scanner for example).  To avoid this, and DRY up the inclusion logic, what do you think about having the subclass invoke a method to request the data pull?